modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java

```diff
@@ -178,7 +186,7 @@ public ServiceOrderDetails getOrderDetailsByOrderId(UUID orderId) {
      */
     public ServiceOrderDetails getLatestModificationOrder(UUID serviceId) {
         ServiceOrderEntity query = new ServiceOrderEntity();
-        query.setServiceId(serviceId);
+        query.setDeployServiceEntity(deployServiceStorage.findDeployServiceById(serviceId));
```

Comment:
do we really need to set the full object in query?
probably it is enough to set `query.setServiceId(deployServiceStorage.findDeployServiceById(serviceId).getId());`